.github/actions/create-pr/action.yml

```diff
@@ -0,0 +1,82 @@
+name: Create a PR if one doesn't exists
+description: >
+  Creates a commit with the current changes to the repo, and opens a PR for that commit. If
+  any PR with the same title exists, then this action is marked as succeeded.
+inputs:
+  commit-message:
+    description: >
+      The message for the commit to be created.
+    required: true
+
+  title:
+    description: >
+      The title of the PR. If empty, the title and body will be determined from the commit message.
+    default: ''
+    required: false
+
+  body:
+    description: >
+      The body (description) of the PR. The `title` input must be specified in order for this input to be used.
+    default: ''
+    required: false
+
+  head-branch:
+    description: >
+      The name of the branch to hold the new commit. If an existing open PR with the same head
+      branch exists, the new branch will be force-pushed to that PR instead of creating a new PR.
+    required: true
+
+  base-branch:
+    description: >
+      The base branch to target with the new PR.
+    required: true
+
+  token:
+    description: |
+      The GitHub token to use. It must have enough privileges to
+      make API calls to create and close pull requests.
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Update git config
+      shell: bash
+      run: |
+        git config --global user.email "github-actions@github.com"
+        git config --global user.name "github-actions[bot]"
+    - name: Commit, Push and Open PR
+      shell: bash
+      env:
+        COMMIT_MESSAGE: ${{ inputs.commit-message }}
+        HEAD_BRANCH: ${{ inputs.head-branch }}
+        BASE_BRANCH: ${{ inputs.base-branch }}
+        GH_TOKEN: ${{ inputs.token }}
```

Comment:
I wonder if this needs to change to `GITHUB_TOKEN` so that the git commands can work.

Reply:
I think, the checkout step has already taken care of the ambiguity. Both `GH_TOKEN` and `GITHUB_TOKEN` work 😄 - I also think there might actually be no need for this since the `GH_TOKEN` is not being used in this action at all...

Reply:
Yes, but this is after the checkout which sets up the whole `gh` cli stuff right?

Reply:
I don't think so. I believe the `checkout` step sets up Git, and uses the token information to set up Git remotes and credentials, but does not do `gh auth` for the CLI. I believe the CLI will still be expecting a token in `GH/GITHUB_TOKEN` and will attempt interactive auth if that's missing (which you don't want in CI).

Reply:
Cool, I'll keep it as it is then 😄